lhotse/dataset/source_separation.py

```diff
@@ -0,0 +1,34 @@
+from pathlib import Path
+from typing import Dict
+
+import torch
+from torch.utils.data import Dataset
+
+from lhotse.cut import CutSet
+from lhotse.utils import Pathlike
+
+
+class SourceSeparationDataset(Dataset):
+    def __init__(self, sources_set: CutSet, mixtures_set: CutSet, root_dir: Pathlike):
+        super().__init__()
+        self.sources_set = sources_set
+        self.mixtures_set = mixtures_set
+        self.cut_ids = list(self.sources_set.cuts.keys())
+        self.root_dir = Path(root_dir)
+        assert set(self.cut_ids) == set(self.mixtures_set.cuts.keys())
+
+    def __getitem__(self, cut_id: str) -> Dict[str, torch.Tensor]:
+        mixture_cut = self.mixtures_set.cuts[cut_id]
+        source_cuts = [self.sources_set.cuts[id_] for id_ in mixture_cut.ingredient_ids]
+        mixture = torch.from_numpy(mixture_cut.load_features(root_dir=self.root_dir))
+        sources = torch.stack(
+            [torch.from_numpy(source_cut.load_features(root_dir=self.root_dir)) for source_cut in source_cuts],
+            dim=0
+        )
```

Comment:
What features would you expect here? 
The sum of the sources's features will probably not sum to the mixture right? 
If we'd like another type of supervision, for example a mask or something like this, we'd compute on the fly or use the `supervisions` from `CutSet`?

Reply:
I think that maybe is it related to `overlay_fbank` ? You can just mix fbanks with no log no ?

Reply:
I looked at one of the datasets in Asteroid, LibriMix, and saw that you were returning a list of sources as well as the mix, so I mimicked it for the first take. We can make this (or another Dataset variant) return a mask instead if you'd like. Can you point me to an example of how do you obtain a mask as supervision?

Reply:
I intended to stick to log-mel energies for now, unless it doesn't make sense for source separation?

Reply:
> I intended to stick to log-mel energies for now, unless it doesn't make sense for source separation?

It's unusual to perform separation on log-mel indeed, but we can try.
In Asteroid, we only feed the waveforms at the ouput of the Dataset, features are computed on the fly, as well as training targets. 
[That](https://github.com/mpariente/asteroid/blob/master/egs/wsj0-mix/DeepClustering/train.py#L129) would be one place where we compute binary masks and ratio masks. 
Granted, this is not the most efficient as those computation could be done in masked time but in this case, we chose to do it in the model rather than in the dataset

Reply:
We can add masks and support spectrograms in a future PR